### PR TITLE
Add Command Event

### PR DESCRIPTION
**Pull request by Rushmead**
This adds the command event, also added an example to the test mod.

### CODE REVIEW

File: patches/net/minecraft/command/CommandHandler.java.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- /net/minecraft/command
++++ /net/minecraft/command
+@@ -81,6 +70,10 @@
+    }
+ 
+    protected boolean tryExecute(ICommandSender var1, String[] var2, ICommand var3, String var4) {
++
++      String[] arguments = xyz.openmodloader.event.impl.CommandEvent.onCommandRan(var2, var3, var1);
++      if(arguments == null){ return false; }
++      var2 = arguments;
```
Comment (shadowfacts) [inline patches/net/minecraft/command/CommandHandler.java.patch]: These two lines can be reduced to one:

```java
If (arguments == null) return false; else var2 = arguments;
```

Comment (Rushmead) [inline patches/net/minecraft/command/CommandHandler.java.patch]: Alrighty, i'll do that :)

File: patches/net/minecraft/command/CommandHandler.java.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+--- /net/minecraft/command
++++ /net/minecraft/command
+@@ -81,6 +70,10 @@
+    }
+ 
+    protected boolean tryExecute(ICommandSender var1, String[] var2, ICommand var3, String var4) {
++
```
Comment (iLexiconn) [inline patches/net/minecraft/command/CommandHandler.java.patch]: no

File: patches/net/minecraft/command/CommandHandler.java.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
++      String[] arguments = xyz.openmodloader.event.impl.CommandEvent.onCommandRan(var2, var3, var1);
++      if(arguments == null) return false; else var2 = arguments;
++
```
Comment (iLexiconn) [inline patches/net/minecraft/command/CommandHandler.java.patch]: no

File: src/main/java/xyz/openmodloader/event/impl/CommandEvent.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    /**
+     * Hook to make related patches much cleaner.
+     * @param args The arguments used in the command
+     * @param command The command ran
+     * @param sender The sender who ran the command
+     * @return The message to actually display.
+     */
+    public static String[] onCommandRan(String[] args, ICommand command, ICommandSender sender) {
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/CommandEvent.java]: 'handle'

File: src/main/java/xyz/openmodloader/test/OMLTestMod.java [hunk 5]
```diff
@@ -34,7 +33,11 @@ public void onEnable() {
 
         OpenModLoader.INSTANCE.EVENT_BUS.register(ScreenshotEvent.class, this::onScreenshot);
 
+
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/test/OMLTestMod.java]: why?

File: src/main/java/xyz/openmodloader/test/OMLTestMod.java [hunk 6]
```diff
@@ -92,6 +95,10 @@ private void onHarvestDrops(BlockEvent.HarvestDrops event){
         event.getDrops().add(new ItemStack(Blocks.DIRT));
     }
 
+    private void onCommandRan(CommandEvent event) {
+        OpenModLoader.INSTANCE.LOGGER.info("Player: " + event.getSender().getName() + " ran command: " + event.getCommand().getCommandName() + " with arguments: " + Arrays.toString(event.getArgs()));
+        event.setCanceled(true);
+    }
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/test/OMLTestMod.java]: and now you're missing one

**Comment by iLexiconn** [comment]
You didn't fix the all, lol

**Pull request merged**